tests/test_fourier.py

```diff
@@ -0,0 +1,47 @@
+from altairexpress import fourier_transform as ft
+import pytest
+import pandas as pd
+import numpy as np
+import altair as alt
+
+
+@pytest.fixture
+def test_fourier():
+    """
+    Tests the fourier function fourier_transform to make sure the outputs are correctly rendering.
+    Returns:
+    --------
+    None
+        The test should pass and no asserts should be displayed.
+    """
+    #Creating test DataFrame objects
+    N = 600
+    T = 1.0 / 800.0
+    x = np.linspace(0.0, N*T, N)
+    y1 = np.sin(50.0 * 2.0*np.pi*x) + 0.5*np.sin(80.0 * 2.0*np.pi*x)
+    y2 = np.sin(40.0 * 2.0*np.pi*x) + 0.5*np.sin(70.0 * 2.0*np.pi*x)
+    my_df1 = pd.DataFrame(data = {'X1': x, 'X2': y1})
+    my_df2 = pd.DataFrame(data = {'X1': x, 'X2': y2})
+    
+    # Test 1
+    assert isinstance(ft.fourier_transform(data = my_df1, time_col = 'X1', data_col = 'X2'), \
+    alt.vegalite.v3.api.Chart), "Return type is not an Altair object!"
+    assert isinstance(ft.fourier_transform(data = my_df2, time_col = 'X1', data_col = 'X2'), \
+    alt.vegalite.v3.api.Chart), "Return type is not an Altair object!"
+    # Test 2
+    assert ft.fourier_transform(data = my_df1, time_col = 'X1', data_col = 'X2') == "line", \
+    "Return plot should be a line plot!"
+    assert ft.fourier_transform(data = my_df2, time_col = 'X1', data_col = 'X2') == "line", \
+    "Return plot should be a line plot!"
+    # Test 3
+    assert ft.fourier_transform(my_df1, 'X1', 'X2').to_dict()['encoding']['x']['type'] == "quantitative", \
+    "Plot x-axis should be of quantitative type!"
+    assert ft.fourier_transform(my_df2, 'X1', 'X2').to_dict()['encoding']['x']['type'] == "quantitative", \
+    "Plot x-axis should be of quantitative type!"
+    # Test 4
+    assert ft.fourier_transform(my_df1, 'X1', 'X2').to_dict()['encoding']['y']['type'] == "quantitative", \
+    "Plot y-axis should be of quantitative type!"
+    assert ft.fourier_transform(my_df2, 'X1', 'X2').to_dict()['encoding']['y']['type'] == "quantitative", \
+    "Plot y-axis should be of quantitative type!"
+
```

Comment:
Make sure that the tests cover all branches (e.g., if your function has an if statement, you should have tests where the if statement is true, and where the if statement is false.)

You should write tests for the following statements in your function:
```
   assert isinstance(time_col, str), 'Column name for time data should be a string!'
   assert isinstance(data_col, str), 'Column name for signal data should be a string!'
   assert isinstance(data, pd.DataFrame), 'Data should be in pandas.DataFrame format!'
   assert((data[data_col] == data[data_col].dropna()).all()), \
     '`NaN` values found in signal column! Please Fill!'
   assert((data[time_col] == data[time_col].dropna()).all()), \
     '`NaN` values found in time column! Please Fill!'
   assert np.isclose(my_time[n + 1] - my_time[n], sampling_freq), \
         'Sampling frequency is not uniformly! Assure that time between samples is constant!'
```

Reply:
Write tests for my tests? Will implement if I have time